docs/Development-Guides/Errors/Error-Types.md

```diff
@@ -25,6 +25,8 @@ Name | Finch Code | Code | Description
 `authentication_error` | `reauthenticate_user` | 401 | The user will need to reconnect their employment system. See more here.
 `invalid_client_error` | | 401 | The provided application credentials were incorrect. Relevant to the `/auth/token` endpoint.
 `unauthorized_request_error` | | 401 | The `access_token` is missing from the header.
+`insufficient_permission_error` | `account_update_required` | 403 | The authenticating account lacks sufficient permissions to perform an operation. The account owner must take manual action to gain sufficient permissions.
+`insufficient_permission_error` | `no_valid_accounts` | 403 | We could not find any suitable connections associated with the authenticating account.
```

Comment:
I wonder if we should be careful with this one. There are other adapters that are throwing a `no_valid_accounts` error but it'll be an `authentication_error`. Should I add another entry in the docs or should I update all instances of `no_valid_accounts` to be an `insufficient_permission`

Reply:
Maybe we can mark the auth error one as `scheduled to deprecate` or something and add a new entry for the it under permission error? Also side note that I thought `no_valid_accounts` means that we cannot find any companies to select after auth but looks like it means we have no access to a valid account in code...

Reply:
> Also side note that I thought no_valid_accounts means that we cannot find any companies to select after auth but looks like it means we have no access to a valid account in code...

Could you elaborate on the difference?

Reply:
Ah wait sorry I think what I said was super misleading. What I mean is that I found a few adapters (like [paylocity](https://github.com/Finch-API/api-server/blob/d6b83addbf034cc57b12edc9075c63ddd4bfbffa/adapters/paylocity/base.js#L600)) that's throwing `no_valid_accounts` when we should throw `account_update_required ` instead.

Reply:
Yea the paylocity one should be `account_update_required`. Did you just know that paylocity was like that off the top of your head? Or did you go through and check a bunch of adapters?

Reason I'm asking - Should we do that type of an audit of our adapters to see where else `no_valid_accounts` is being used incorrectly?

Reply:
I found it during working on the paylocity adapter. Also did a quick check on the other adapters using `no_valid_accounts` and most of them are used correctly. Might be good to do a thorough audit

Reply:
Okay that makes sense, thanks! Here's a [ticket](https://app.asana.com/0/1202163450897194/1202275893368249) for us to do that audit. We can wait on merging these docs changes for now since there are already a lot of errors that have gone undocumented.